src/components/Input/index.js

```diff
@@ -214,7 +214,14 @@ export const Input = React.forwardRef(
         : !validate
     const showResetButton =
       (props.value && props.value.length && props.onReset) ||
-      (!props.value && ref && props.onReset)
+      (!props.value &&
+        ref &&
+        props.onReset &&
+        ((ref &&
+          ref.current &&
+          ref.current.value &&
+          ref.current.value.length) ||
+          false))
```

Comment:
Bit complex now this part. Maybe add a short comment explaining what’s happening? Controlled with value, uncontrolled with ref and value, etc.

Reply:
Force pushed a change, let me know what you think.